src/data/types.ts

```diff
@@ -90,6 +92,130 @@ export interface StreamInfo {
   viewer_count: number;
 }
 
+export interface StreamAnnouncementInfo {
+  id: string;
+  category_name: string;
+  language: string;
+  started_at: Date;
+  thumbnail_url: string;
+  title: string;
+  streamer_info: StreamerInfo;
+  viewer_count: number;
+  streaming_service: StreamingService;
+  online: boolean;
+  archivedStream?: VideoByUserIdResponse;
+}
+
+
+export interface StreamingService {
+  name: string;
+  base_url: string;
+}
+
+export const Glimesh: StreamingService = {
+  name: "Glimesh",
+  base_url: "https://glimesh.tv/"
+} as const
+
+export const Twitch: StreamingService = {
+  name: "Twitch",
+  base_url: "https://twitch.tv/"
+} as const
+
+export interface StreamAnnouncementInfoResolver {
+  resolve(): Promise<StreamAnnouncementInfo>;
+}
+export class TwitchStreamInfoResolver
+  implements StreamAnnouncementInfoResolver
+{
+  constructor(private streamInfo: StreamInfo) {}
+
+  async resolve(): Promise<StreamAnnouncementInfo> {
+    const streamer = await UserManager.getUserAsStreamerInfoById(this.streamInfo.user_id);
+    let category = await fetchGameById(this.streamInfo.game_id);
+    if (!category) {
+      category = { name: "" };
+    }
+    const started_at = new Date(this.streamInfo.started_at);
+
+    return {
+      id: this.streamInfo.id,
+      title: this.streamInfo.title,
+      category_name: category.name,
+      language: this.streamInfo.language,
+      started_at: started_at,
+      thumbnail_url: this.streamInfo.thumbnail_url,
+      viewer_count: this.streamInfo.viewer_count,
+      streamer_info: streamer,
+      streaming_service: Twitch,
+      online:true
+    };
+  }
+}
+
+export class GlimeshStreamInfoProvider
+  implements StreamAnnouncementInfoResolver
+{
+  constructor(private streamInfo: GlimeshStreamInfo) {} // private streamInfo: GlimeshStreamInfo) {}
+  async resolve(): Promise<StreamAnnouncementInfo> {
```

Comment:
Not sure I'm entirely comfortable with this being async when it doesn't need to be, but I do understand that it needs to do that because of the Twitch version, so probably nothing that we can do about this.

Reply:
It hurts my eyes a bit too but I don't know any better way to do it